decidim-core/lib/tasks/upgrade/user_groups_migration.rake

```diff
@@ -111,6 +120,34 @@ namespace :decidim do
           puts "===== Skipped, no entries found to transform."
         end
       end
+
+      desc "Remove notification entries related with user groups"
+      task remove_groups_notifications: :environment do
+        items = Decidim::Notification.where(
+          event_class: %w(
+            Decidim::Comments::CommentByFollowedUserGroupEvent
+            Decidim::Comments::UserGroupMentionedEvent
+            Decidim::DemotedMembershipEvent
+            Decidim::InvitedToGroupEvent
+            Decidim::JoinRequestAcceptedEvent
+            Decidim::JoinRequestCreatedEvent
+            Decidim::JoinRequestRejectedEvent
+            Decidim::PromotedToAdminEvent
+            Decidim::RemovedFromGroupEvent
+            Decidim::UserGroupAdminEvent
+            Decidim::UserGroupCreatedEvent
+            Decidim::UserGroupUpdatedEvent
+          )
+        )
+
+        if (count = items.count).positive?
+          items.destroy_all
```

Comment:
This may cause a memory issue, as it is, you would load all the notifications, then destroy one of it to ensure the callbacks are being ran. 

maybe a simple  thing like this would help ?
```suggestion
          items.find_each(&:destroy)
```

Reply:
Oh, yes I missed that important detail